java/gorpc/src/main/java/com/youtube/vitess/gorpc/codecs/bson/BsonClientCodec.java

```diff
@@ -45,7 +43,7 @@ public void WriteRequest(Request request, Object args) throws IOException {
   public void ReadResponseHeader(Response response) throws IOException {
     BSONObject headerBson = decoder.readObject(socket.getInputStream());
     response.setServiceMethod(new String((byte[]) headerBson.get(Constants.SERVICE_METHOD)));
-    response.setSeq(UnsignedLong.fromLongBits((long) headerBson.get(Constants.SEQ)));
+    response.setSeq((long)headerBson.get(Constants.SEQ));
```

Comment:
I pushed a new build to the mvn-repo. Let me know if it still doesn't work. I think you should be able to revert this Java change. LGTM on the other changes though.